libc/string/string_tokpbrkstr.c

```diff
@@ -0,0 +1,687 @@
+	TEST_ASSERT_EQUAL_STRING("sit", token);
+
+	TEST_ASSERT_EQUAL_PTR(&str1[22], (token = strtok_r(NULL, separators, &restState1)));
+	TEST_ASSERT_EQUAL_STRING("amet", token);
+	TEST_ASSERT_EQUAL_PTR(&str2[22], (token = strtok_r(NULL, separators, &restState2)));
+	TEST_ASSERT_EQUAL_STRING("amet", token);
+
+	TEST_ASSERT_EQUAL_PTR(&str1[28], (token = strtok_r(NULL, separators, &restState1)));
+	TEST_ASSERT_EQUAL_STRING("consectetur", token);
+	TEST_ASSERT_EQUAL_PTR(&str2[28], (token = strtok_r(NULL, separators, &restState2)));
+	TEST_ASSERT_EQUAL_STRING("consectetur", token);
+
+	TEST_ASSERT_EQUAL_PTR(&str1[40], (token = strtok_r(NULL, separators, &restState1)));
+	TEST_ASSERT_EQUAL_STRING("adipiscing", token);
+	TEST_ASSERT_EQUAL_PTR(&str2[40], (token = strtok_r(NULL, separators, &restState2)));
+	TEST_ASSERT_EQUAL_STRING("adipiscing", token);
+
+	TEST_ASSERT_EQUAL_PTR(&str1[51], (token = strtok_r(NULL, separators, &restState1)));
+	TEST_ASSERT_EQUAL_STRING("elit", token);
+	TEST_ASSERT_EQUAL_PTR(&str2[51], (token = strtok_r(NULL, separators, &restState2)));
+	TEST_ASSERT_EQUAL_STRING("elit", token);
+
+	TEST_ASSERT_EQUAL_PTR(&str1[57], (token = strtok_r(NULL, separators, &restState1)));
+	TEST_ASSERT_EQUAL_STRING("Maecenas", token);
+	TEST_ASSERT_EQUAL_PTR(&str2[57], (token = strtok_r(NULL, separators, &restState2)));
+	TEST_ASSERT_EQUAL_STRING("Maecenas", token);
+
+	TEST_ASSERT_EQUAL_PTR(&str1[66], (token = strtok_r(NULL, separators, &restState1)));
+	TEST_ASSERT_EQUAL_STRING("eleifend", token);
+	TEST_ASSERT_EQUAL_PTR(&str2[66], (token = strtok_r(NULL, separators, &restState2)));
+	TEST_ASSERT_EQUAL_STRING("eleifend", token);
+
+	TEST_ASSERT_EQUAL_PTR(&str1[75], (token = strtok_r(NULL, separators, &restState1)));
+	TEST_ASSERT_EQUAL_STRING("elementum", token);
+	TEST_ASSERT_EQUAL_PTR(&str2[75], (token = strtok_r(NULL, separators, &restState2)));
+	TEST_ASSERT_EQUAL_STRING("elementum", token);
+
+	TEST_ASSERT_EQUAL_PTR(&str1[85], (token = strtok_r(NULL, separators, &restState1)));
+	TEST_ASSERT_EQUAL_STRING("tellum", token);
+	TEST_ASSERT_EQUAL_PTR(&str2[85], (token = strtok_r(NULL, separators, &restState2)));
+	TEST_ASSERT_EQUAL_STRING("tellum", token);
+
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok_r(NULL, separators, &restState1));
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok_r(NULL, separators, &restState2));
+#endif
+}
+
+
+TEST(string_tok_r, torn)
+{
+#ifdef __phoenix__
+	TEST_IGNORE();
+#else
+	char *rest = NULL,
+		 *tok,
+		 str[] = "Lor\0em ipsum";
+
+	const char *separators = " ";
+
+	TEST_ASSERT_EQUAL_PTR(str, (tok = strtok_r(str, separators, &rest)));
+	TEST_ASSERT_EQUAL_STRING("Lor", tok);
+	TEST_ASSERT_EQUAL_PTR(NULL, (tok = strtok_r(NULL, separators, &rest)));
+#endif
+}
+
+
+TEST(string_tok_r, empty_args)
+{
+#ifdef __phoenix__
+	TEST_IGNORE();
+#else
+	char empty[] = { 0 },
+		 *rest;
+
+	/* Different scenarios of acquiring null pointer as output*/
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok_r("", "d", &rest));
```

Comment:
The definition of strtok_r says it must be a pointer that is not initialized to ensure it will be assigned to the correct user-defined place:
![image](https://github.com/phoenix-rtos/phoenix-rtos-tests/assets/49757239/625e1f09-c697-44c4-adcf-a5cf6be8fc36)
Application use also says:
![image](https://github.com/phoenix-rtos/phoenix-rtos-tests/assets/49757239/fceb72d6-33b1-4d16-b013-b9859114b572)